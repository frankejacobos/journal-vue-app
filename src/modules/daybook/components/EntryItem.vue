<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: entry.id } })"
  >
    <div class="entry-title d-flex">
      <span class="text-success fs-4 fw-bold">{{ day }}</span>
      <span class="mx-1 fs-5">{{ month }}</span>
      <span class="mx-2 fw-light">{{ year }}</span>
    </div>
    <div class="entry-description">
      {{ shortText }}
    </div>
  </div>
</template>

<script>
import getDayMonthYear from "../helpers/getDayMonthYear";
export default {
  props: {
    entry: {
      type: Object,
      required: true,
    },
  },
  computed: {
    day() {
      const { day } = getDayMonthYear(this.entry.date);
      return day;
    },
    month() {
      const { month } = getDayMonthYear(this.entry.date);
      return month;
    },
    year() {
      const { year } = getDayMonthYear(this.entry.date);
      return year;
    },
    shortText() {
      return this.entry.text.substring(0, 100);
    },
  },
};
</script>

<style lang="scss" scoped>
.entry-container {
  background-color: #fff;
  border-radius: 5px;
  box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.1);
  transition: all 0.2s ease-in;
  &:hover {
    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.2);
  }
  .entry-description {
    padding: 10px;
    font-size: 14px;
  }
}
</style>
