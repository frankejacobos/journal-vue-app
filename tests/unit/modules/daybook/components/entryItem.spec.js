import { shallowMount } from "@vue/test-utils";
import EntryItem from "@/modules/daybook/components/EntryItem.vue";
import { journalState } from "@/../tests/unit/mocks/test-journal-state";

describe("Pruebas en el componente EntryItem", () => {
  const mockRouter = { push: jest.fn() };
  test("Debe de hacer match con el snapshot", () => {
    const wrapper = shallowMount(EntryItem, {
      props: {
        entry: journalState.entries[0],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("debe redireccionar con click en el entry-container", () => {
    const wrapper = shallowMount(EntryItem, {
      props: { entry: journalState.entries[0] },
      global: { mocks: { $router: mockRouter } },
    });
    wrapper.find(".entry-container").trigger("click");
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: journalState.entries[0].id },
    });
  });
  test("pruebas en las propiedades computadas", () => {
    const wrapper = shallowMount(EntryItem, {
      props: { entry: journalState.entries[0] },
      global: { mocks: { $router: mockRouter } },
    });
    expect(wrapper.vm.day).toBe(1);
    expect(wrapper.vm.month).toBe("Enero");
    expect(wrapper.vm.year).toBe("2020, Mi√©rcoles");
  });
});
