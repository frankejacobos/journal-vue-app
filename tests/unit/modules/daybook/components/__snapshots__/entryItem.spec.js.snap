// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente EntryItem Debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-4 fw-bold">1</span><span class="mx-1 fs-5">Enero</span><span class="mx-2 fw-light">2020, Mi√©rcoles</span></div>
  <div class="entry-description">Entry 1</div>
</div>
`;
